File: 9663.py
import numpy as np

col = np.zeros(15)
N = int(input())
cnt = 0

def dfs(row):
    global cnt, N
    if(row > N):
        cnt += 1
        return
    else:
        for i in range(1, N + 1):
            col[row] = i
            if(isPromising(row)): dfs(row + 1)
            else: col[row] = 0

def isPromising(row):
    for i in range(1, row):
        if(col[i] == col[row]):return False
        if(abs(col[i] - col[row]) == abs(i -row)):return False
    return True

for i in range(1, N + 1):
    col = np.zeros(15)
    col[1] = i
    dfs(2)

print(cnt)

File: 9663.cpp
#include <iostream>
#include <cstring>

using namespace std;

int N, col[15], cnt = 0;

bool isPromising(int row) {
	for (int i = 1; i < row; i++) {
		if (col[i] == col[row]) return false;
		if (abs(col[i] - col[row]) == abs(i - row))return false;
	}
	return true;
}

void dfs(int row) {
	if (row > N) cnt++;
	else {
		for (int i = 1; i <= N; i++) {
			col[row] = i;
			if (isPromising(row)) dfs(row + 1);
			else col[row] = 0;
		}
	}
}

int main() {
	cin >> N;
	for (int i = 1; i <= N; i++) {
		memset(col, 0, sizeof(col));
		col[1] = i;
		dfs(2);
	}
	cout << cnt << endl;
	return 0;
}